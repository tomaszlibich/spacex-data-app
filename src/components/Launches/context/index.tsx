import { createContext } from "react";
import { LaunchesContextType } from "../types";

export const initialData: LaunchesContextType = {
  launches: [],
  sorting: null,
  setSorting: () => {},
  search: undefined,
  setSearch: () => {},
};

export const LaunchesContext = createContext(initialData);
