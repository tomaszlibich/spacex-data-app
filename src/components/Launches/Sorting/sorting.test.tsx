import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { LaunchesContext, initialData } from "../context";
import { LaunchesSorting } from "./index";

configure({ adapter: new Adapter() });

const setSortingSpy = jest.fn();
const initialContextData = { ...initialData, setSorting: setSortingSpy };

let component;
let tree: any;
let wrapper: any;

describe("LaunchesSorting", () => {
  describe('When component is rendered', () => {
    beforeEach(() => {
      component = renderer.create(
        <LaunchesSorting />,
      );
      tree = component.toJSON();
    });

    it("should match the snapshot", () => {
      expect(tree).toMatchSnapshot();
    });

    describe('And no sorting has been applied', () => {
      beforeEach(() => {
        wrapper = mount(
          <LaunchesContext.Provider value={{ ...initialContextData }}>
            <LaunchesSorting />
          </LaunchesContext.Provider>
        )
      });

      it("should render only one item on the list", async () => {
        expect(setSortingSpy).not.toHaveBeenCalled();
      });

      describe('And sorting by NAME has been applied', () => {
        beforeEach(() => {
          wrapper = mount(
            <LaunchesContext.Provider value={{ ...initialContextData }}>
              <LaunchesSorting />
            </LaunchesContext.Provider>
          )

          wrapper.find('button:first-child').simulate('click');
        });

        it("should set the correct sorting for the list", async () => {
          expect(setSortingSpy).toHaveBeenCalledWith('name');
        });
      });

      describe('And sorting by DATE has been applied', () => {
        beforeEach(() => {
          wrapper = mount(
            <LaunchesContext.Provider value={{ ...initialContextData }}>
              <LaunchesSorting />
            </LaunchesContext.Provider>
          )

          wrapper.find('button:last-child').simulate('click');
        });

        it("should set the correct sorting for the list", async () => {
          expect(setSortingSpy).toHaveBeenCalledWith('date');
        });
      });
    });
  });
});
