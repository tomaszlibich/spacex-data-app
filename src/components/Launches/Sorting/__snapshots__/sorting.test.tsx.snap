// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchesSorting When component is rendered should match the snapshot 1`] = `
<div
  className="sorting"
>
  <button
    className=""
    onClick={[Function]}
  >
    Sort by name
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    Sort by date
  </button>
</div>
`;
